src/libs/antares/study/scenario-builder/applyToMatrix.hxx

```diff
@@ -89,4 +96,48 @@ bool ApplyToMatrix(uint& errors,
 
     return ret;
 }
+
+template<class StringT, class D>
+bool ApplyToMatrixPowerCredits(uint& errors,
```

Comment:
CheckValidity has already specialization for DataSeriesHydro class.
If we specialize applyToMatrix function, even then we will have a problem because DataSeriesHydroClass has a two matrix for time series numbers. I don't see implementing this on proper way without some refactoring of other classes that are calling these template functions. Maybe we can open discussion for this one?